packages/cli/api-importers/openapi/openapi-v2-importer/src/3.1/paths/RequestBodyConverter.ts

```diff
@@ -94,6 +92,176 @@ export class RequestBodyConverter extends AbstractConverter<
+                )
+            };
+        } else {
+            const requestBody = HttpRequestBody.reference({
+                contentType,
+                docs: this.requestBody.description,
+                requestBodyType: convertedSchema.type
+            });
+
+            return {
+                requestBody,
+                inlinedTypes: Object.fromEntries(
+                    Object.entries(convertedSchema.inlinedTypes).filter(([key]) => key !== schemaId)
+                )
+            };
+        }
+    };
+
+    private convertRequestBodyProperty({
+        context,
+        property,
+        contentType
+    }: {
+        context: OpenAPIConverterContext3_1;
+        property: ObjectProperty;
+        contentType: string;
+    }) {
+        if (context.isFile(property.valueType)) {
+            return FileUploadRequestProperty.file(
+                FileProperty.file({
+                    key: property.name,
+                    isOptional: false,
+                    contentType
+                })
+            );
+        } else if (
+            context.isOptional(property.valueType) &&
+            // @ts-expect-error: TS2339 we know this is an optional typeReference.
+            context.isFile(property.valueType.container.optional)
+        ) {
+            return FileUploadRequestProperty.file(
+                FileProperty.file({
+                    key: property.name,
+                    isOptional: true,
+                    contentType
+                })
+            );
+        } else if (
+            context.isList(property.valueType) &&
+            // @ts-expect-error: TS2339 we know this is an list typeReference.
+            context.isFile(property.valueType.container.list)
+        ) {
+            return FileUploadRequestProperty.file(
+                FileProperty.fileArray({
+                    key: property.name,
+                    isOptional: false,
+                    contentType
+                })
+            );
+        } else if (
+            context.isList(property.valueType) &&
+            // @ts-expect-error: TS2339 we know this is an list typeReference.
+            context.isOptional(property.valueType.container.list) &&
+            // @ts-expect-error: TS2339 we know this is an optional typeReference.
+            context.isFile(property.valueType.container.list.container.optional)
+        ) {
+            return FileUploadRequestProperty.file(
+                FileProperty.fileArray({
+                    key: property.name,
+                    isOptional: false,
+                    contentType
+                })
+            );
+        } else if (
+            context.isOptional(property.valueType) &&
+            // @ts-expect-error: TS2339 we know this is an optional typeReference.
+            context.isList(property.valueType.container.optional) &&
+            // @ts-expect-error: TS2339 we know this is an list typeReference.
```

Comment:
hmm `// @ts-expect-error` this is suspect, we shouldn't ever need to allow a ts-error